src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+) extends FgBioTool with LazyLogging {
+
+  import Cigar._
+
+  Io.assertReadable(Seq(callVcf, truthVcf))
+  intervals.foreach(Io.assertReadable)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+    // Setup
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    // check the sequence dictionaries.
+    val (dict, sampleNames) = {
+      val callReader = new VCFFileReader(callVcf.toFile, true)
+      val truthReader = new VCFFileReader(truthVcf.toFile, true)
+      callReader.getFileHeader.getSequenceDictionary.assertSameDictionary(truthReader.getFileHeader.getSequenceDictionary)
+      callReader.close()
+      truthReader.close()
+      import scala.collection.JavaConversions.collectionAsScalaIterable
+      (callReader.getFileHeader.getSequenceDictionary, Seq(truthReader, callReader).map(_.getFileHeader.getSampleNamesInOrder.head))
+    }
+
+    val intervalList = intervals.map { intv => IntervalList.fromFile(intv.toFile) }
+
+    // get the phased blocks
+    logger.info("Gathering the phased blocks for the call sample")
+    val callPhaseBlockDetector: OverlapDetectorByContig = {
+      val callReader  = new VCFFileReader(callVcf.toFile, true)
+      val detector = PhaseBlock.toOverlapDetector(callIterator(callReader), dict=callReader.getFileHeader.getSequenceDictionary)
+      callReader.close()
+      detector
+    }
+    logger.info("Gathering the phased blocks for the truth sample")
+    val truthPhaseBlockDetector: OverlapDetectorByContig = {
+      val truthReader = new VCFFileReader(truthVcf.toFile, true)
+      val detector = PhaseBlock.toOverlapDetector(truthIterator(truthReader, intervalList=intervalList), dict=truthReader.getFileHeader.getSequenceDictionary)
+      truthReader.close()
+      detector
+    }
+
+    // get an iterator of the pairs
+    val callReader  = new VCFFileReader(callVcf.toFile, true)
+    val truthReader = new VCFFileReader(truthVcf.toFile, true)
+    val pairedIterator = {
+      new JointVariantContextIterator(
+        iters = Seq(truthIterator(truthReader, intervalList=intervalList), callIterator(callReader)),
+        dict = dict,
+        sampleNames = sampleNames
+      )
+    }
+
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+    // Create the phasing cigar
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+    logger.info("Creating the phasing CIGAR")
+    val metric = new AssessPhasingMetric()
+    val cigar = toCigar(
+      pairedIterator          = pairedIterator,
+      truthPhaseBlockDetector = truthPhaseBlockDetector,
+      callPhaseBlockDetector  = callPhaseBlockDetector,
+      metric                  = metric,
```

Comment:
Both need the same information at the site level, so it is convenient to both at the same time.   Perhaps the method name should be changed.